From: Simon McVittie <smcv@debian.org>
Date: Fri, 2 Aug 2019 20:13:39 +0100
Subject: Disable an optimization when building with gcc 9

This appears to break the clutter autopkgtest, but I don't know why.

Signed-off-by: Simon McVittie <smcv@debian.org>
Bug: https://bugs.debian.org/931921
---
 glib/gtestutils.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/glib/gtestutils.c b/glib/gtestutils.c
index 8817890..b22b0d4 100644
--- a/glib/gtestutils.c
+++ b/glib/gtestutils.c
@@ -1598,6 +1598,9 @@ void
     test_built_files_dir = test_argv0_dirname;
 }
 
+#if defined(__GNUC__) && __GNUC__ >= 9 && !defined(__clang__)
+__attribute__((optimize("no-tree-pre")))
+#endif
 static void
 test_run_seed (const gchar *rseed)
 {
